instrumentation/redisson-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/RedissonClientTracer.java

```diff
@@ -24,24 +28,33 @@ protected String getInstrumentationName() {
   }
 
   @Override
-  protected String normalizeQuery(Object args) {
+  protected String normalizeQuery(Object command) {
     // get command
-    if (args instanceof CommandsData) {
-      List<CommandData<?, ?>> commands = ((CommandsData) args).getCommands();
+    if (command instanceof CommandsData) {
+      List<CommandData<?, ?>> commands = ((CommandsData) command).getCommands();
       StringBuilder commandStrings = new StringBuilder();
-      for (CommandData commandData : commands) {
-        commandStrings.append(commandData.getCommand().getName()).append(";");
+      for (CommandData<?, ?> commandData : commands) {
+        commandStrings.append(normalizeSingleCommand(commandData)).append(";");
       }
       if (commandStrings.length() > 0) {
         commandStrings.deleteCharAt(commandStrings.length() - 1);
       }
       return commandStrings.toString();
-    } else if (args instanceof CommandData) {
-      return ((CommandData) args).getCommand().getName();
+    } else if (command instanceof CommandData) {
+      return normalizeSingleCommand((CommandData<?, ?>) command);
     }
     return "Redis Command";
   }
 
+  private static String normalizeSingleCommand(CommandData<?, ?> command) {
+    List<String> args = new ArrayList<>();
+    if (command.getCommand().getSubName() != null) {
+      args.add(command.getCommand().getSubName());
+    }
+    args.addAll(Stream.of(command.getParams()).map(String::valueOf).collect(Collectors.toList()));
```

Comment:
Can `args` be `List<Object>`? This would eliminate this stream and thus extra allocations.

Reply:
Done.